tests/ignite/metrics/test_multilabel_confusion_matrix.py

```diff
@@ -0,0 +1,438 @@
+    return
+
+
+def test_simple_ND_input():
+
+    # 3D tests
+    num_iters = 100
+    num_samples = 100
+    num_classes = 10
+    size_3d = 4
+
+    for _ in range(num_iters):  # 3D tests
+        target = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d))
+        prediction = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d))
+        mlcm = MultiLabelConfusionMatrix(num_classes, normalized=False)
+        mlcm.update([torch.tensor(prediction), torch.tensor(target)])
+        ignite_CM = mlcm.compute().numpy()
+        target_reshaped = target.transpose(0, 2, 1).reshape(size_3d * num_samples, num_classes)
+        prediction_reshaped = prediction.transpose(0, 2, 1).reshape(size_3d * num_samples, num_classes)
+        sklearn_CM = multilabel_confusion_matrix(target_reshaped, prediction_reshaped)
+        assert np.all(sklearn_CM.astype(np.int64) == ignite_CM.astype(np.int64))
+
+    size_4d = 4
+    for _ in range(num_iters):  # 4D tests
+        target = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d, size_4d))
+        prediction = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d, size_4d))
+        mlcm = MultiLabelConfusionMatrix(num_classes, normalized=False)
+        mlcm.update([torch.tensor(prediction), torch.tensor(target)])
+        ignite_CM = mlcm.compute().numpy()
+        target_reshaped = target.transpose(0, 2, 3, 1).reshape(size_3d * size_4d * num_samples, num_classes)
+        prediction_reshaped = prediction.transpose(0, 2, 3, 1).reshape(size_3d * size_4d * num_samples, num_classes)
+        sklearn_CM = multilabel_confusion_matrix(target_reshaped, prediction_reshaped)
+        assert np.all(sklearn_CM.astype(np.int64) == ignite_CM.astype(np.int64))
+
+    size_5d = 4
+    for _ in range(num_iters):  # 5D tests
+        target = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d, size_4d, size_5d))
+        prediction = np.random.randint(0, 2, size=(num_samples, num_classes, size_3d, size_4d, size_5d))
+        mlcm = MultiLabelConfusionMatrix(num_classes, normalized=False)
+        mlcm.update([torch.tensor(prediction), torch.tensor(target)])
+        ignite_CM = mlcm.compute().numpy()
+        target_reshaped = target.transpose(0, 2, 3, 4, 1).reshape(
+            size_3d * size_4d * size_5d * num_samples, num_classes
+        )
+        prediction_reshaped = prediction.transpose(0, 2, 3, 4, 1).reshape(
+            size_3d * size_4d * size_5d * num_samples, num_classes
+        )
+        sklearn_CM = multilabel_confusion_matrix(target_reshaped, prediction_reshaped)
+        assert np.all(sklearn_CM.astype(np.int64) == ignite_CM.astype(np.int64))
+
+    return
+
+
+def test_simple_batched():
+
+    num_iters = 100
+    num_samples = 100
+    num_classes = 10
+    batch_size = 1
+
+    for _ in range(num_iters):
+        mlcm = MultiLabelConfusionMatrix(num_classes, normalized=False)
+        target = np.random.randint(0, 2, size=(1, num_classes))
```

Comment:
Nit: maybe, we can simply generate n batches and iterate over that data instead of concat:
```python
targets = np.random.randint(0, 2, size=(num_iters, batch_size, num_classes))
predictions = np.random.randint(0, 2, size=(num_iters, batch_size, num_classes))
for target, prediction in zip(targets, predictions):
    ...
```

Reply:
So, instead of concatenating I am now generating matrices once in every iteration of the outer loop.